lib/bolt/plugin.rb

```diff
@@ -167,6 +163,20 @@ def initialize(config, pal, analytics, load_plugins: true)
       end
       @unresolved_plugin_configs['puppetdb'] = config.puppetdb if config.puppetdb
       @plugin_hooks = DEFAULT_PLUGIN_HOOKS.dup
+      @unresolved_plugin_hooks = config.plugin_hooks
```

Comment:
This seems like it should be a boolean instead of data, and you can just use `resolve_references(config.plugin_hooks)` instead of `resolve_references(unresolved_plugin_hooks)`.

Reply:
👍 Good suggestion. Opted for lazily merging the config and assigning it to `@plugin_hooks`.